models/DataObject/ClassDefinition/Data/Numeric.php

```diff
@@ -545,6 +559,39 @@ public function getForCsvExport($object, $params = [])
         return (string)$data;
     }
 
+    /**
+     * returns sql query statement to filter according to this data types value(s)
+     *
+     * @param string $value
+     * @param string $operator
+     * @param array $params optional params used to change the behavior
+     *
+     * @return string
+     */
+    public function getFilterConditionExt($value, $operator, $params = [])
+    {
+        $db = \Pimcore\Db::get();
+        $name = $params['name'] ?: $this->name;
+        $key = $db->quoteIdentifier($name);
+        if (!empty($params['brickPrefix'])) {
+            $key = $params['brickPrefix'].$key;
+        }
+
+        if ($value === 'NULL') {
+            if ($operator === '=') {
+                $operator = 'IS';
+            } elseif ($operator === '!=') {
+                $operator = 'IS NOT';
+            }
+        }
+
+        if (!is_array($value) && !is_object($value) && in_array($operator, self::$validFilterOperators)) {
```

Comment:
Wouldn't this be better?

```suggestion
        if (is_scalar($value) && in_array($operator, self::$validFilterOperators, true)) {
```

Then `$value` may only be `int`, `string`, `float` or `bool`, but **not** `array`, `object`, `resource` or `null`.

Reply:
copied from https://github.com/pimcore/pimcore/blob/10.5/models/DataObject/ClassDefinition/Data.php#L581

Reply:
maybe we should only allow `int` and `float`?

Reply:
it probably can be a `string` as well, right? 
What about using `is_numeric()` that should cover all cases?